GDSCommon-Demo/GDSCommon-DemoTests/TextInputViewControllerTests.swift

```diff
@@ -0,0 +1,138 @@
+@testable import GDSCommon
+@testable import GDSCommon_Demo
+import XCTest
+
+final class TextInputViewControllerTests: XCTestCase {
+    typealias InputType = Bool
+    var sut: TextInputViewController<InputType>!
+    var viewModel: (any TextInputViewModel)?
+    
+    var resultAction: ((Bool) -> Void)!
+    
+    var didSetResult: [Bool] = [false]
+    var screenDidAppear: Bool = false
+    var didTapButton = false
+    var didDismissScreen = false
+    
+    override func setUp() {
+        super.setUp()
+
+        screenDidAppear = false
+        
+        resultAction = { bool in
+            self.didSetResult.append(bool)
+        }
+        
+        let textFieldViewModel = MockTextFieldViewModel<InputType>()
+        
+        let viewModel = MockTextInputViewModel<InputType>(textFieldViewModel: textFieldViewModel) { _ in
+            
+        } appearAction: {
+            
```

Comment:
do we maybe want print statements in here?

Reply:
I'd missed adding the test properties being set here. I could probably add a couple more unit tests tbh. Will have a look.